installer/pkg/nextgen/asset.go

```diff
@@ -0,0 +1,6 @@
+package nextgen
+
+type Asset interface {
+	Dependencies() []Asset
```

Comment:
So you explain a little more how the assetstore might use the hash from Dependencies to actually generate the corr asset.

Reply:
Dependencies() defines the assets it requires. asset store ensures that those assets are in state before calling generate on the *it* asset.

Reply:
I missed the discussion related to using a Merkle DAG for the asset store. Could someone elaborate on how we plan on using the properties of a Merkle DAG? Maybe this could be added to the design document.